### PR TITLE
gk: check IP support before lookup

### LINKED ISSUE #259: GK block does not check IP support before lookup
After receiving a burst of packets on both the front and back interface, the GK block extracts the packet information fully and performs a FIB lookup without checking whether it supports the IP type of the packet.

### PR DESCRIPTION
**Pull request by Hrish1**
The PR closes #259

### CODE REVIEW

File: gk/main.c [hunk 1]
```diff
@@ -1383,6 +1385,16 @@ process_pkts_front(uint16_t port_front, uint16_t port_back,
 			continue;
 		}
 
+		if (packet.flow.proto == ETHER_TYPE_IPv4 &&
+				 !ipv4_configured_front) {
+			drop_packet_front(pkt, instance);
```
Comment (mengxiang0811) [inline gk/main.c]: This ``if`` statement for IPv4 packets and the below statement for IPv6 packets are the same, so we can combine the conditions check to one, and the code becomes simpler.

The same comment applies to the back interface.

**Comment by mengxiang0811** [comment]
I cannot add comments there. The commit message should be in the same format as the title of this patch (i.e., ``[block_name]: one sentence description of the patch``), so suggestion: change the commit message to be ``gk: check IP support before lookup``.

File: gk/main.c (hunk 1 above)
Comment (AltraMayor) [reply gk/main.c]: Once we have a single `if` statement, please wrap the test with `unlikely()`.

**Comment by Hrish1** [comment]
@mengxiang0811 @AltraMayor I have updated the PR with the changes

File: gk/main.c [hunk 2]
```diff
@@ -1383,6 +1385,14 @@ process_pkts_front(uint16_t port_front, uint16_t port_back,
 			continue;
 		}
 
+		if (unlikely(packet.flow.proto == ETHER_TYPE_IPv4 &&
+						 !ipv4_configured_front ||
+						 packet.flow.proto == ETHER_TYPE_IPv6 &&
+						 !ipv6_configured_front)) {
```
Comment (mengxiang0811) [inline gk/main.c]: Style: the above three lines have too many tabs, please take a look at the [dpdk coding style](https://doc.dpdk.org/guides/contributing/coding_style.html#c-indentation). Note that, ``Line length is recommended to be not more than 80 characters, including comments.``

File: gk/main.c [hunk 3]
```diff
@@ -1653,6 +1665,14 @@ process_pkts_back(uint16_t port_back, uint16_t port_front,
 			continue;
 		}
 
+		if (unlikely(packet.flow.proto == ETHER_TYPE_IPv4 &&
+						 !ipv4_configured_back ||
+						 packet.flow.proto == ETHER_TYPE_IPv6 &&
+						 !ipv6_configured_back)) {
```
Comment (mengxiang0811) [inline gk/main.c]: Style: see above comments.

**Comment by mengxiang0811** [comment]
I cannot comment there. Seems that you are changing ``dependencies/bird`` sub-module, is it expected? if so, why?

**Comment by Hrish1** [comment]
I think I staged it by mistake.  My bad!

**Comment by Hrish1** [comment]
@mengxiang0811 made the styling changes.

**Comment by AltraMayor** [comment]
Repeating @mengxiang0811's request:
I cannot add comments there. The commit message should be in the same format as the title of this patch (i.e., ``[block_name]: one sentence description of the patch``), so suggestion: change the commit message to be ``gk: check IP support before lookup``.

Mine: add `This patch closes #259` to the message of the patch. This help tracking changes while using `git log`.

**Comment by Hrish1** [comment]
This patch closes #259

**Comment by AltraMayor** [comment]
The message `This patch closes #259` must be part of the message of the patch in Git. Put another way, after the title, `gk: check IP support before lookup`, you leave a bank line, and on the third line write `This patch closes #259`. Take a look at the output of `git log` for examples.

**Comment by Hrish1** [comment]
@AltraMayor Done!

**Comment by AltraMayor** [comment]
I just merged your pull request, @Hrish1. Thank you.

**Pull request merged**